### DISCUSSION TITLE
Couldn't connect to custom domains after setting up SMTP and IMAP

### DISCUSSION THREAD

**Discussion by elhananjair** [the-djmaze/snappymail#1738] [Q&A] [answered]
Greetings guys, 
I have tried to set up both SMTP and IMAP on my server running Nextcloud. I am trying to connect to any mail servers easily without adding custom domains in snappymail admin setting. 

I have installed **postfix** and **Dovecot** already, both are running as the service seems fine. 

```
● dovecot.service - Dovecot IMAP/POP3 email server
     Loaded: loaded (/usr/lib/systemd/system/dovecot.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Tue 2024-09-03 21:44:52 EAT; 1s ago
       Docs: man:dovecot(1)
             https://doc.dovecot.org/
   Main PID: 135091 (dovecot)
     Status: "v2.3.21.1 (d492236fa0) running"
      Tasks: 4 (limit: 9348)
     Memory: 5.6M (peak: 5.7M)
        CPU: 49ms
     CGroup: /system.slice/dovecot.service
             ├─135091 /usr/sbin/dovecot -F
             ├─135092 dovecot/anvil
             ├─135093 dovecot/log
             └─135094 dovecot/config

Sep 03 21:44:52 smartbits.dev systemd[1]: Starting dovecot.service - Dovecot IMAP/POP3 email server...
Sep 03 21:44:52 smartbits.dev dovecot[135091]: master: Dovecot v2.3.21.1 (d492236fa0) starting up for imap, pop3, lmtp
Sep 03 21:44:52 smartbits.dev systemd[1]: Started dovecot.service - Dovecot IMAP/POP3 email server.

● postfix.service - Postfix Mail Transport Agent
     Loaded: loaded (/usr/lib/systemd/system/postfix.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Sun 2024-09-01 22:43:21 EAT; 1 day 23h ago
    Process: 65032 ExecStartPre=/usr/sbin/restorecon -R /var/spool/postfix/pid (code=exited, status=0/SUCCESS)
    Process: 65033 ExecStartPre=/usr/libexec/postfix/aliasesdb (code=exited, status=0/SUCCESS)
    Process: 65036 ExecStartPre=/usr/libexec/postfix/chroot-update (code=exited, status=0/SUCCESS)
    Process: 65038 ExecStart=/usr/sbin/postfix start (code=exited, status=0/SUCCESS)
   Main PID: 65106 (master)
      Tasks: 4 (limit: 9348)
     Memory: 5.6M (peak: 9.5M)
        CPU: 3.119s
     CGroup: /system.slice/postfix.service
             ├─ 65106 /usr/libexec/postfix/master -w
             ├─ 65108 qmgr -l -t unix -u
             ├─ 65730 tlsmgr -l -t unix -u
             └─133643 pickup -l -t unix -u

Sep 01 22:43:20 domain systemd[1]: Starting postfix.service - Postfix Mail Transport Agent...
Sep 01 22:43:21 domain postfix[65104]: postfix/postlog: starting the Postfix mail system
Sep 01 22:43:21 domain postfix/postfix-script[65104]: starting the Postfix mail system
Sep 01 22:43:21 domain postfix/master[65106]: daemon started -- version 3.8.5, configuration /etc/postfix
Sep 01 22:43:21 domain systemd[1]: Started postfix.service - Postfix Mail Transport Agent.
Sep 01 22:56:41 domain postfix/smtpd[65728]: connect from unknown[::1]
Sep 01 22:56:41 domain postfix/smtpd[65728]: disconnect from unknown[::1] ehlo=2 starttls=1 quit=1 commands=4
```

But when I try to add an account on snappymail, I get  **Authentication Failed** error message although the password I am using is the correct one. 

**Here is the screenshot from snappymail admin settings**
![image](https://github.com/user-attachments/assets/e9617935-618e-46e5-a69a-812eeeee2b06)

**And the error showing up as I try to add an account**
![image](https://github.com/user-attachments/assets/e7372d84-0163-4851-b10b-463ce5a2608f)

```
Sep 01 22:44:21 domain auth[65179]: pam_unix(dovecot:auth): check pass; user unknown
Sep 01 22:44:21 domainauth[65179]: pam_unix(dovecot:auth): authentication failure; logname= uid=0 euid=0 tty=dovecot ruser=user@mydomain rhost=::1
Sep 01 22:44:26 domain dovecot[39858]: imap-login: Disconnected: Connection closed (auth failed, 1 attempts in 5 secs): user=<user@mydomain>, method=PLAIN, rip=::1, lip=::1, TLS, session=<fVEFDRQhkN0AAAAAAAAAAAAAAAAAAAAB>

```

Can someone please support me on this? 
Thanks

**Comment by cm-schl** [comment]
Hi,
What do you mean by "try to add an account"? Does your first login is successful and only when you're trying to add additional accounts you're getting this error?
Does your server uses usernames like user@example.com or only "user"?

**Reply by elhananjair** [reply]
Hello there thanks for replying
I have already connected my Gmail account to snappymail by generating an app password and using it. But now I am looking for a feature to connect any kind of domain to snappymail, and for that, I need to install SMTP and IMAP serving applications. 

I haven't configured anything I just installed both and the configuration files are on default status, I didn't change anything. I am trying to add account with a custom domain, just like  **name@domain.com**
Can you please tell me where should I configure additional things that are required by snappymail?

**Reply by cm-schl** [reply]
Hi,

in order to send/receive mails from a local server you have to make sure if and how you can login to your server outside of SnappyMail .
Is a login with a username `name@domain.com` possible or do you have to use just `name` as username?
If you're able to check your mails outside from SnappyMail you can be sure that the imap service is running correctly.

Afterwards you can create a new domain entry inside the SnappyMail admin section: https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#menu-domains . I have to say i never used the "*" config like you did in your screenshot, so maybe this could work or interfere with your gmail configuration you already have. So maybe use a domain name as described in the documentation 🙂.
When everything is configured you can try the "test" function here to see if SnappyMail is configured correctly: https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#testing-your-configuration

**Reply by elhananjair** [reply]
Hello
I didn't want to limit domains on snappymail that's why I am trying to use *. I like the users to connect their preferred mail address to snappymail easily with any domain.

Since there is gmail domain defined already it is working fine. Still, I am trying to add another account which based on a domain from a different website, and for that I need to install SMTP and IMAP server which I did already. Still, I couldn't understand which parameters must be changed on SMTP or IMAP service to connect snappymail.

**Reply by cm-schl** [reply]
SnappyMail itself is "just" an IMAP client that accesses your IMAP server... So have you tried to connect to your server using an other client like Thunderbird?

**Reply by elhananjair** [reply]
Nope, I haven't tested with another one, but I still suspect that it could be a configuration issue on both IMAP and SMTP. I didn't touch both configuration files, since I couldn't find a guideline on that and what was required by snappymail.

**Reply by elhananjair** [reply]
@the-djmaze any support on this, please? I wanted to know if what I was trying to achieve was impossible.

**Reply by the-djmaze** [reply, marked as answer]
Use the `Login Autoconfig` extension

> I like the users to connect their preferred mail address to snappymail easily with any domain.

**Reply by elhananjair** [reply]
> > Hello I didn't want to limit domains on snappymail that's why I am trying to use *. I like the users to connect their preferred mail address to snappymail easily with any domain.
> 
> Use the `Login Autoconfig` extension

Thank you for replying, I have enabled the `Login Autoconfig`  extension and tried to add an account to snappymail but have another issue now, `Request Timeout`. 
Here is a log from Nextcloud 

```
{"reqId":"ZuSI-dflGvnd9pNu2KYdVwAAxQU","level":3,"time":"2024-09-13T18:50:31+00:00","remoteAddr":"196.190.62.169","user":"USERNAME","app":"PHP","method":"POST","url":"/apps/snappymail/?/Json/&q[]=/0/","message":"file_get_contents(http://autodiscover.MYDOMAIN/autodiscover/autodiscover.xml Linux x86_64; rv:130.0) Gecko/20100101 Firefox/130.0","version":"29.0.5.1","data":{"app":"PHP"}
```

**Reply by elhananjair** [reply]
@the-djmaze UPDATE
It worked fine for all domains except for an email address created with a domain I am using for Nextcloud. It shows `Request Timeout` on that specific domain, but other domains work fine. any clues?

**Reply by the-djmaze** [reply]
`http://autodiscover.MYDOMAIN` is not resolvable and timeouts.
You need to create your NextCloud domain config manually.

**Reply by elhananjair** [reply]
Thank you

**Comment by elhananjair** [comment]
One thing to mention here is, that the Nextcloud Mail application works fine without additional configuration. You can log in with any domain you like.

**Reply by the-djmaze** [reply]
Not really, it also fails for my custom IMAP setups.
Difference is that NC mail uses autoconfig by default and SnappyMail does not for security reasons.

So if you are the company IT and need to secure mail access...
SnappyMail is secure by default and NC Mail is not.

I also could not find a setting in NC Mail to disable autoconfig and/or manual config to disallow users from using private/other mail accounts.

Also see:
- https://help.nextcloud.com/t/disable-setting-add-new-mail-account/106160
- https://github.com/nextcloud/mail/issues/5759

So it all depends on the security requirements for a NextCloud instance in use.

**Reply by elhananjair** [reply]
Thank you for the brief explanation
